channels/rdpsnd/pulse/rdpsnd_pulse.c

```diff
@@ -449,10 +449,12 @@ static void rdpsnd_pulse_start(rdpsndDevicePlugin* device)
 {
 	rdpsndPulsePlugin* pulse = (rdpsndPulsePlugin*)device;
 
+	pa_threaded_mainloop_lock(pulse->mainloop);
```

Comment:
If pulse->stream isn't set wouldn't the following return keep the mainloop locked?
I'd assume that the right place for the lock would be after the check.